### DISCUSSION TITLE
Is it possible to render Chart.js ?

### DISCUSSION THREAD

**Discussion by raed667** [vercel/satori#347] [Q&A]
I want to render a Chart.js canvas based chart on the server. Then return the image on Next JS edge runtime.

I tried SSR existing [solutions](https://github.com/SeanSobey/ChartjsNodeCanvas) for Chart.js, but I'm having issues with canvas.

Is this something doable at all or am I wasting my time?

**Comment by shuding** [comment]
Nope, this is not doable. `canvas` isn't a supported element in Satori. But I think you can probably try to make d3.js work if you managed to run in on the server.

**Comment by joulev** [comment]
You can render to a canvas normally, then convert it to a PNG, then render the PNG with a data URL in an `<img>` tag in Satori.

https://github.com/SeanSobey/ChartjsNodeCanvas/issues/121#issuecomment-2071392998
